nservicebus/azure-service-bus/configuration/configuration.md

```diff
@@ -0,0 +1,200 @@
+---
+title: Full Azure Service Bus Configuration API
+summary: List of all Azure Service Bus Transport configuration settings
+tags:
+- Azure
+- Cloud
+- Configuration
+---
+
+## Full Configuration API
+
+The full configuration API can be accessed from the `UseTransport<AzureServiceBusTransport>()` extension method and provides low level access to almost all aspects of the transport's behavior.
+ 
+## Configuring The Topology
+ 
+A topology defines what the underlying layout of Azure Service Bus messaging entities looks like, which ones are used and how they relate to each other. There are 2 topologies built in into the transport `EndpointOrientedTopology` and `ForwardingTopology`.
+
+ * `UseTopology<T>()`: Specifies which topology should be used by the transport, instantiating it using the default constructor.
+ * `UseTopology<T>(Func<T>)`: Specifies which topology should be used by the transport, instantiating it using a factory method.
+ * `UseTopology<T>(T)`: Specifies which topology should be used by the transport, instance created by the calling code.
+ 
+### Endpoint Oriented Topology
+
+The endpoint oriented topology defines a queue and a topic per endpoint. This model causes the side effect that subscribing endpoints also need to know which publishing endpoints are responsible for publishing a given event type. The following configuration settings allow to define this relationship.
+
+ * `RegisterPublisherForType(string, Type)`: Registers the publishing endpoint for a given type.
+ * `RegisterPublisherForAssembly(string, Assembly)`: Registers the publishing endpoint for all types in an assembly.
+
+### Forwarding Topology
+
+The forwarding topology defines a queue per endpoint and a shared set of topics to do the publishing. The Azure Service Bus forwarding feature is used to forward messages from subscriptions on the shared set of topics to the input queue of each endpoint. As forwarding cannot be used in conjunction with partitioning, NServiceBus will perform the partitioning instead using a bundle of topics. 
```

Comment:
If we link to Sean's doco, then I think we can skip description or limit it to 1 sentence.